utils/globalvar_test.go

```diff
@@ -0,0 +1,62 @@
+package utils
+
+import (
+	"os"
+	"testing"
+)
+
+func TestParseIntEnvVar(t *testing.T) {
+	tests := []struct {
+		name         string
+		varName      string
+		varValue     string
+		defaultValue int
+		expected     int
+		expectError  bool
+	}{
+		{
+			name:         "environment variable not set",
+			varName:      "TEST_ENV_VAR_NOT_SET",
+			defaultValue: 10,
+			expected:     10,
+			expectError:  false,
+		},
+		{
+			name:         "environment variable set with valid integer",
+			varName:      "TEST_ENV_VAR_VALID",
+			varValue:     "20",
+			defaultValue: 10,
+			expected:     20,
+			expectError:  false,
+		},
+		{
+			name:         "environment variable set with invalid integer",
+			varName:      "TEST_ENV_VAR_INVALID",
+			varValue:     "invalid",
+			defaultValue: 10,
+			expected:     10,
+			expectError:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.varValue != "" {
+				os.Setenv(tt.varName, tt.varValue)
+			} else {
+				os.Unsetenv(tt.varName)
+			}
+
+			val, err := parseIntEnvVar(tt.varName, tt.defaultValue)
+			if tt.expectError && err == nil {
+				t.Fatalf("Expected an error, got nil")
+			} else if !tt.expectError && err != nil {
+				t.Fatalf("Expected no error, got %v", err)
+			}
+
+			if val != tt.expected {
+				t.Fatalf("Expected value %d, got %d", tt.expected, val)
```

Comment:
issue: test case does not use the conventional `testify/assert`.

Did ChatGPT or something of the sort write this? We commonly use `testify/assert`, its assertions are simpler than that, so there’s no need for ifs and `t.Fatalf`.


```go
tests := []struct{
  name string
  inputVarName string
  inputVarValue string
  defaultValue int
  expectedValue int
  expectedError error
}{
  for _, tc := range tests {
    if tc.varValue != "" {
      t.Setenv(tt.varName, tt.varValue)
    } else {
      t.Setenv(tt.varName, "")
    }

    val, err := parseIntEnvVar(tt.varName, tt.defaultValue)

    assert.Equalf(t, expectedError, gotError)
    assert.Equalf(t, expectedValue, gotValue)
  }
}
```